subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/attributes/ExclusiveVariantsIntegrationTest.groovy

```diff
@@ -0,0 +1,213 @@
+        given:
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            configurations {
+                sample1 {
+                    canBeResolved = false
+                    canBeConsumed = true
+
+                    attributes {
+                        attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage, Usage.JAVA_RUNTIME))
+                        attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category, "custom"))
+                    }
+
+                    outgoing {
+                        // Force identical capabilities by overriding the default
+                        capability('org.gradle:sample:1.0')
+                    }
+                }
+
+                sample2 {
+                    canBeResolved = false
+                    canBeConsumed = true
+
+                    attributes {
+                        attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage, Usage.JAVA_RUNTIME))
+                        attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category, "custom"))
+                    }
+
+                    outgoing {
+                        // Force identical capabilities by overriding the default
+                        capability('org.gradle:sample:1.0')
+                    }
+                }
+            }""".stripIndent()
+
+        expect:
+        executer.expectDocumentedDeprecationWarning("Consumable configurations with identical capabilities within a project must have unique attributes, but configuration ':sample2' and configuration ':sample1' contain identical attribute sets.")
+        succeeds("outgoingVariants")
+    }
+
+    def "the default capability never triggers warning, even if there is a match"() {
+        given:
+        settingsFile << "rootProject.name = 'sample'"
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            group = 'org.gradle'
+            version = '1.0'
+
+            configurations {
+                sample1 {
+                    canBeResolved = false
+                    canBeConsumed = true
+
+                    attributes {
+                        attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage, Usage.JAVA_RUNTIME))
+                        attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category, "custom"))
+                    }
+                }
+
+                sample2 {
+                    canBeResolved = false
+                    canBeConsumed = true
+
+                    attributes {
+                        attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage, Usage.JAVA_RUNTIME))
+                        attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category, "custom"))
+                    }
+
+                    outgoing {
+                        // Force identical capabilities as the default
+                        capability('org.gradle:sample:1.0')
```

Comment:
I think this should create the warning. An explicit capability that equals the default is equivalent.

Reply:
OK.  I will add the default capability to the set that is compared if there are no other capabilities.